src/java/org/apache/cassandra/streaming/messages/StreamMessage.java

```diff
@@ -48,6 +48,11 @@ public static void serialize(StreamMessage message, DataOutputStreamPlus out, in
         message.type.outSerializer.serialize(message, out, version, session);
     }
 
+    public static long serializedSize(StreamMessage message, int version) throws IOException
+    {
+        return 1 + message.type.outSerializer.serializedSize(message, version);
+    }
+
     public static StreamMessage deserialize(ReadableByteChannel in, int version, StreamSession session) throws IOException
```

Comment:
Ah now I get it. Streaming originally used ReadableByteChannel which is inherently a copying API. Super weird code here though it's doing non-blocking reads?